hyco-https/lib/HybridConnectionHttpsServer.js

```diff
@@ -573,52 +576,60 @@ function controlChannelRequest(server, message) {
   // handler to call when the connection sequence completes
   var self = server;
   
-  if ( message.request.method) {
+  if (message.request.method) {
+    // we received a GET request or a small POST http request
+    // the response should be sent over the control channel
     req = new IncomingMessage(message, server.controlChannel);
     if ( message.request.body == true) {
        self.pendingRequest = req;
     } else {
       req.push(null);
     }
-  }
-  
-  // execute the web socket rendezvous with the server
-  try {
-    var client = new WebSocket(address, {
-      headers: headers,
-      perMessageDeflate: false
-    });
-    client.on('open', function() {
-      
-      server.emit('requestchannel', client);
-      if (self.options.clientTracking) {
-        self.clients.push(client);
-        ws.on('close', function() {
-          var index = self.clients.indexOf(client);
-          if (index != -1) {
-            self.clients.splice(index, 1);
-          }
-        });
-      }
 
-      // do we have a request or is this just rendezvous?
-      if ( message.request.method) {
-         var res = new ServerResponse(req);
-         res.requestId = message.request.id;
-         res.assignSocket(client);
-         server.emit('request', req, res);
-      }
-    });
+    var res = new ServerResponse(req);
```

Comment:
>ServerResponse [](start = 18, length = 14)

Can you add a test where there's a small request over the control channel and the response body is > 64KB? Doesn't that need a rendezvous to occur? I can't tell where that code is.

Reply:
Added the test. It goes through the "else" block below which creates a new web socket.

---
In reply to: [315880613](https://github.com/Azure/azure-relay-node/pull/33#discussion_r315880613) [](ancestors = 315880613)

Reply:
which test is that?

---
In reply to: [315951489](https://github.com/Azure/azure-relay-node/pull/33#discussion_r315951489) [](ancestors = 315951489,315880613)